Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -120,11 +123,82 @@ StyledMarkedText::Style StyledMarkedText::computeStyleForUnmarkedMarkedText(cons
     return style;
 }
 
+static Vector<StyledMarkedText> coalesceAdjacentWithSameRanges(const Vector<StyledMarkedText>& styledTexts)
+{
+    Vector<StyledMarkedText> frontmostMarkedTexts;
+    frontmostMarkedTexts.append(styledTexts[0]);
```

Comment:
is it possible for `styledTexts` to be empty?

Reply:
Don't think so, I set isHighlights to true if markedTexts is not empty and contains highlights. Thus, if it gets to this function call, it is because isHighlights is true. But to be safe I could add another check within the function, what do you think?

Reply:
I think either an `ASSERT(!styledTexts.isEmpty)` or a `if (styledTexts.isEmpty) { ASSERT_NOT_REACHED(); return { } }` would be good, just in case!